### ISSUE TITLE
Support for [AKS Node Image Upgrade outside of k8s version upgrade]

### ISSUE THREAD

**Issue by asubmani** [hashicorp/terraform-provider-azurerm#20171]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Issue Support for [14600](https://github.com/hashicorp/terraform-provider-azurerm/issues/14600) was closed by @tombuildsstuff . 
Most likely the feature ask was misunderstood.

AKS allows upgrading the NODE OS IMAGE of a nodepool **INDEPENDENT** of a kubernetes minor/patch upgrade. 

AKS product group releases a new OS image ~ every 15 days.  Consider the below scenario:

AKS Control Plane and nodePools upgraded to 1.23 on "2022.10.17" . NODE OS IMAGE = `AKSUbuntu-1804gen2containerd-2022.10.17`
There have been a few Node OS  Image updates since:

1. [AKSUbuntu-1804gen2containerd-2022.10.17](https://github.com/Azure/AKS/blob/2023-01-08/vhd-notes/aks-ubuntu/AKSUbuntu-1804/2022.10.24.txt)
2. [AKSUbuntu-1804gen2containerd-2022.10.24](https://github.com/Azure/AKS/blob/2023-01-08/vhd-notes/aks-ubuntu/AKSUbuntu-1804/2022.11.01.txt)
3. [AKSUbuntu-1804gen2containerd-2022.11.01](https://github.com/Azure/AKS/blob/2023-01-08/vhd-notes/aks-ubuntu/AKSUbuntu-1804/2022.11.01.txt)
4. AKSUbuntu-1804gen2containerd-2023.01.10 

All of these are aplicable to any version of 1.23.x  or 1.24.x  

For Clusters running 1.25 there is a different nodeImage with Ubuntu22.

Currently the only option is to create a newNodePool with the same kubnernetes version and then manually drain the old nodepool. (If using terraform)

Here is the az cli command that upgrades a nodepool without upgrading the kubernetes version
`az aks nodepool upgrade -g $RG --cluster-name $AKS --name genericqa1 --node-image-only`

It would be great to allow upgrading node image updates and track that.

### New or Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster_node_pool,azurerm_kubernetes_cluster

### Potential Terraform Configuration

```hcl
resource "azurerm_kubernetes_cluster_node_pool" "example" {
  name                  = "internal"
  kubernetes_cluster_id = azurerm_kubernetes_cluster.example.id
  vm_size               = "Standard_DS2_v2"
  node_count            = 1
  node_image            = latest | AKSUbuntu-1804gen2containerd-2023.01.10 


}


resource "azurerm_kubernetes_cluster" "example" {
  name                = "example-aks1"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  dns_prefix          = "exampleaks1"

  default_node_pool {
    name       = "defaultsystem"
    node_count = 1
    vm_size    = "Standard_D2_v2"
    node_image            = latest | AKSUbuntu-1804gen2containerd-2023.01.10 
  }
```

### References

https://github.com/hashicorp/terraform-provider-azurerm/issues/14600

**Comment by aristosvo** [comment]
To support this request, here a reference to the [REST API docs](https://learn.microsoft.com/en-us/rest/api/aks/agent-pools/upgrade-node-image-version?tabs=HTTP)

**Comment by asubmani** [comment]
>Hi @aristosvo ,

Thanks for your reply. 

I need my to be at the latest node image version, "on-demand". Setting the `automatic_channel_upgrade`  to `node-image` will always (and automatically) update the `node-image`.

I want to run a `terraform plan/apply` that will talk to the API and:

1. see if the current available node-image is newer than the node-image that is deployed at present.
2.  if yes, then show this as `~ 1 to update..`  and run the plan to update the node image to latest version.

AFAIK, the azurerm provider is NOT aware of the current deployed node-image version OR the latest version available to upgrade to.

**Comment by asubmani** [comment]
I can choose to skip a few `node-images` simply with an `ignore_lifecycle_changes` or by not running the `apply`. Consider the below scenarios:
1. Cluster and nodepools deployed on 1st Jan 2023. `node-image == AKSUbuntu-1804gen2containerd-2023.01.01`
2. AKS releases `node-image: AKSUbuntu-1804gen2containerd-2023.01.10`  ## plan reports this, but I don't update.
3. AKS releases `node-image: AKSUbuntu-1804gen2containerd-2023.01.19` ## Plan again reports this as a change (~ 1 update) ## This time I choose to `apply`. I also have the flexibility to slectively do an `ignore_lifecycle_changes` on specific `node-pools` where I WANT to retain the older image.

`automatic_channel_upgrade` is a switch you turn on/off to allow AKS product group to automatically manage `node-images` per node-pool.  

This should be avoided in PRODUCTION clusters as some times certain images may have breaking changes.

**Comment by asubmani** [comment]
For the time being, even using `azure/azapi` provider would be good.

**Comment by stephybun** [comment]
Support for this was merged in #22187 and #21760 so should be possible from v3.63.0 of the provider onwards.
